Command/ChangePasswordCommand.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * (c) FSi sp. z o.o. <info@fsi.pl>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FSi\Bundle\AdminSecurityBundle\Command;
+
+use FSi\Bundle\AdminSecurityBundle\Event\AdminSecurityEvents;
+use FSi\Bundle\AdminSecurityBundle\Event\ChangePasswordEvent;
+use FSi\Bundle\AdminSecurityBundle\Security\User\UserPasswordChangeInterface;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class ChangePasswordCommand extends ContainerAwareCommand
+{
+    /**
+     * @see Command
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('fsi:user:change-password')
+            ->setDescription('Change password of a user.')
+            ->setDefinition(array(
+                new InputArgument('email', InputArgument::REQUIRED, 'The email'),
+                new InputArgument('password', InputArgument::REQUIRED, 'The password'),
+            ))
+            ->setHelp(<<<EOT
+The <info>fsi:user:change-password</info> command changes user's password:
+
+  <info>php app/console fsi:user:change-password john@example.com</info>
+
+This interactive shell will first ask you for a password.
+
+You can alternatively specify the new password as a second argument:
+
+  <info>php app/console fsi:user:change-password john@example.com mynewpassword</info>
+
+EOT
+            );
+    }
+
+    /**
+     * @see Command
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $email      = $input->getArgument('email');
+        $password   = $input->getArgument('password');
+
+        $userRepository = $this->getContainer()->get('admin_security.repository.user');
+        $user = $userRepository->findUserByEmail($email);
+        if (!($user instanceof UserPasswordChangeInterface)) {
```

Comment:
Also, if this is an interface for the User, I'd name it something like UserWithChangeablePasswordInterface, current name suggests that it's for a class containing the password to be changed or a service doing the change. Perphaps even something like CanChangePasswordInterface would do.